subprojects/docs/src/docs/userguide/migration/upgrading_version_6.adoc

```diff
@@ -76,6 +76,12 @@ taskA.threadPoolSize = taskB.outputFile.map { it.text.toInteger() }
 In this case, calling `get()`, or any other query method, on `taskA.threadPoolSize` will produce a deprecation warning if done prior to `taskB` completing, as the file has not yet been generated.
 This will become an error in Gradle 7.0
 
+==== AbstractCompile and SourceDirectorySet use directory properties to model destinations
```

Comment:
I think this needs a better title that says something more about the effects.  "directory properties" probably doesn't mean much.  e.g.,

> Deprecated properties in AbstractCompile and SourceDirectorySet for changing output destination

Reply:
Ok. Will change in upcoming deprecation PR (for 6.2).